src/tokenizedSubaccounts/BaseOnChainSigningTSA.sol

```diff
@@ -58,6 +58,21 @@ abstract contract BaseOnChainSigningTSA is BaseTSA {
     emit ActionSigned(msg.sender, hash, action);
   }
 
+  function signRfqActionData(IMatching.Action memory action, IRfqModule.TradeData[] memory trades)
+    external
+    virtual
+    onlySigner
+  {
+    bytes32 hash = getActionTypedDataHash(action);
+    IRfqModule.TakerOrder memory takerOrder = abi.decode(action.data, (IRfqModule.TakerOrder));
+    action.data = abi.encode(IRfqModule.RfqOrder(takerOrder.maxFee, trades));
```

Comment:
🔴 Pretty sure this logic actually allows the executor to accept ANY trade, since the action.data.hash (in the MakerOrder) doesn't necessarily match the trade data you pass in along with it. 

Since that's not verified you can pass in a hash of a different trade and then verify against some acceptable "trades" array

Reply:
I ended up hashing the extra data and seeing if that is equal to the `TakerOrder` hash. I think this should work since we do an equivalent hash in the RfqModule. 

```
    IRfqModule.TakerOrder memory takerOrder = abi.decode(action.data, (IRfqModule.TakerOrder));
    if (keccak256(extraData) != takerOrder.orderHash) {
      revert PPT_TradeDataDoesNotMatchOrderHash;
    }